tools/codegen/gen.py

```diff
@@ -859,23 +858,30 @@ def write_sharded(
             env_callable: Callable[[T], Dict[str, List[str]]],
             num_shards: int,
             base_env: Optional[Dict[str, Any]] = None,
+            sharded_keys: Set[str]
     ) -> None:
 
         everything: Dict[str, Any] = {'shard_id': 'Everything'}
         shards: List[Dict[str, Any]] = [{'shard_id': f'_{i}'} for i in range(num_shards)]
+        all_shards = [everything] + shards
 
         if base_env is not None:
-            everything.update(base_env)
-            for shard in shards:
+            for shard in all_shards:
                 shard.update(base_env)
 
+        for key in sharded_keys:
+            for shard in all_shards:
+                if key in shard:
+                    assert isinstance(shard[key], list), "sharded keys in base_env must be a list"
+                    shard[key] = shard[key].copy()
+                else:
+                    shard[key] = []
```

Comment:
This loop ensures the key always exists in the environment, even if none of the functions hashed into that file.

Reply:
Awesome, thanks!